### PR TITLE
Refactor: Clean up `RpcModuleSelection` API

### PR DESCRIPTION
**Pull request by prestwich**
- Use `HashSet` instead of vec internally
- Create statics for `All` and `Standard` hashsets
- Refactor to use `&'static HashSet` where appropriate
- use standard pattern for `as_selection` `to_selection` and `into_selection`
- extend `ConflictingModules` with conflict information
- add `len()`
- condense `are_identical` logic where available
- modify `From` impls on `RpcModuleSelection`

drive-by
- add functions to modify `TransportRpcModuleConfig` when mutable by taking the values
- add accessor for inner config on `TransportRpcModuleConfig`

### CODE REVIEW

File: crates/rpc/rpc-builder/src/lib.rs [hunk 1]
```diff
@@ -616,34 +616,38 @@ pub enum RpcModuleSelection {
...
 
 // === impl RpcModuleSelection ===
 
+/// The standard modules to instantiate by default `eth`, `net`, `web3`
+// TODO: replace with LazyLock
+static STANDARD_SELECTION: OnceLock<HashSet<RethRpcModule>> = OnceLock::new();
+
+/// All modules.
+// TODO: replace with LazyLock
+static ALL_SELECTION: OnceLock<HashSet<RethRpcModule>> = OnceLock::new();
```
Comment (mattsse) [inline crates/rpc/rpc-builder/src/lib.rs:630]: don't think we need a static for this.

this is only used once --all or for help messages so allocating is better than keeping them around

File: crates/rpc/rpc-builder/src/lib.rs [hunk 2]
```diff
@@ -1819,6 +1875,26 @@ impl TransportRpcModuleConfig {
...
+
+    /// Takes the ipc transport configuration, leaving `None` in its place.
+    pub fn take_ipc(&mut self) -> Option<RpcModuleSelection> {
+        self.ipc.take()
+    }
+
+    /// Takes the module configuration, leaving `None` in its place.
+    pub fn take_config(&mut self) -> Option<RpcModuleConfig> {
+        self.config.take()
+    }
```
Comment (mattsse) [inline crates/rpc/rpc-builder/src/lib.rs:1896]: can we replace these with fn (http|ws|ip)_mut -> Option<&mut>

**Review by mattsse — CHANGES_REQUESTED** [review]
can't really remember why this was a vec, but using a map instead is more appropriate.

**Comment by mattsse** [comment]
cc @leruaa

File: crates/rpc/rpc-builder/src/lib.rs (hunk 2 above)
Comment (prestwich) [reply crates/rpc/rpc-builder/src/lib.rs]: ✅

File: crates/rpc/rpc-builder/src/lib.rs (hunk 1 above)
Comment (prestwich) [reply crates/rpc/rpc-builder/src/lib.rs]: kk. this means removing `as_selection`, and extra clones in `are_identical` seems fine tho

**Review by mattsse — APPROVED** [review]
~~lgtm, pedantic clippy~~

**Pull request merged**